### PR TITLE
system channel clean up - msgprocessor package

### PR DESCRIPTION
**Pull request by semil**
1. Cleaned dependencies on system channel from the msgprocessor package.
2. Removed templator usages.

#### Type of change

- Improvement (improvement to code, performance, etc)
- Test update

#### Description

As described in issue #4189.

#### Related issues

Issue: #4189.

### CODE REVIEW

File: orderer/common/msgprocessor/expiration_test.go [hunk 1]
```diff
@@ -22,12 +22,14 @@ import (
...
 //go:generate counterfeiter -o mocks/config_resources.go --fake-name Resources . configResources
 
+// TODO delete?
 type configResources interface {
 	channelconfig.Resources
 }
 
 //go:generate counterfeiter -o mocks/orderer_config.go --fake-name OrdererConfig . ordererConfig
 
+// TODO delete?
```
Comment (tock-ibm) [inline orderer/common/msgprocessor/expiration_test.go]: no, it is an auxiliary interface used to create the mock

File: orderer/common/msgprocessor/expiration_test.go [hunk 2]
```diff
@@ -22,12 +22,14 @@ import (
 
 //go:generate counterfeiter -o mocks/config_resources.go --fake-name Resources . configResources
 
+// TODO delete?
```
Comment (tock-ibm) [inline orderer/common/msgprocessor/expiration_test.go]: no, it is an auxiliary interface used to create the mock

File: orderer/common/msgprocessor/maintenancefilter_test.go [hunk 3]
```diff
@@ -34,6 +34,10 @@ func newMockOrdererConfig(migration bool, state orderer.ConsensusType_State) *mo
 	return mockOrderer
 }
 
+func mockCrypto() *mocks.SignerSerializer {
+	return &mocks.SignerSerializer{}
+}
+
```
Comment (tock-ibm) [inline orderer/common/msgprocessor/maintenancefilter_test.go]: just place `&mocks.SignerSerializer{}` where needed, no need for a method

File: orderer/common/msgprocessor/expiration_test.go (hunk 1 above)
Comment (tock-ibm) [reply orderer/common/msgprocessor/expiration_test.go]: Check the mocks that remain, you'll find that `mocks/sig_filter_support.go` is never used. delete it and also track down the generate statement that created it and remove it as well

File: orderer/common/multichannel/registrar.go [hunk 4]
```diff
@@ -557,11 +556,6 @@ func (r *Registrar) ChannelsCount() int {
...
 }
 
-// NewChannelConfig produces a new template channel configuration based on the system channel's current config.
-func (r *Registrar) NewChannelConfig(envConfigUpdate *cb.Envelope) (channelconfig.Resources, error) {
-	return r.templator.NewChannelConfig(envConfigUpdate)
-}
-
 // CreateBundle calls channelconfig.NewBundle
 func (r *Registrar) CreateBundle(channelID string, config *cb.Config) (channelconfig.Resources, error) {
 	return channelconfig.NewBundle(channelID, config, r.bccsp)
```
Comment (tock-ibm) [inline orderer/common/multichannel/registrar.go]: safe to remove

File: orderer/common/msgprocessor/expiration_test.go (hunk 1 above)
Comment (semil) [reply orderer/common/msgprocessor/expiration_test.go]: Done!

**Review by tock-ibm — APPROVED** [review]

**Pull request merged**